monte_carlo.ipynb

```diff
@@ -183,62 +190,36 @@
     "\n",
     "adjustingFirm = np.array([[min(np.where(timeOfAdjustmentsCompany[simulationNumber]==priceAdjustmentsBeforeHorizon[simulationNumber][i] )[0]) for i in range(numberPriceAdjustmentsBeforeHorizon[simulationNumber])] for simulationNumber in range(numberSimulations)])\n",
     "\n",
-    "randomCustomerScoring = np.array([[0 for _ in range(numberPlayers)] for simulationNumber in range(numberSimulations)])\n",
-    "randomCustomerDecision = np.array([[-1 for _ in range(numberCustomerEvents)] for simulationNumber in range(numberSimulations)])\n",
     "randomCustomerScoringQuality = np.array([[random.uniform(0,1) for _ in range(numberCustomerEvents)] for simulationNumber in range(numberSimulations)])\n",
     "randomCustomerScoringRating = np.array([[random.uniform(0,0.5) for _ in range(numberCustomerEvents)] for simulationNumber in range(numberSimulations)])\n",
-    "\n",
-    "As = np.array([[[0 for _ in range(numberConsideredMarketSituations)]for _ in range(numberSimulations)] for simulationNumber in range(numberPlayers)])\n",
-    "A = np.array([[[float(0) for _ in range(numberConsideredMarketSituations)]for _ in range(6)]for _ in range(numberPlayers)])\n",
-    "\n",
-    "P_ori = np.array([[float(0) for _ in range(len(prices))] for _ in range(numberConsideredMarketSituations)])\n",
-    "P_ori2 = np.array([[float(0) for _ in range(len(prices))] for _ in range(numberConsideredMarketSituations)])\n",
-    "\n",
-    "playersWithoutWatched = [i for i in range(numberPlayers) if i!=watchedCompany]\n",
-    "for w in range(numberConsideredMarketSituations):\n",
-    "    pricesCompaniesPredW= pricesCompaniesPred[:,w]\n",
-    "    qualityCompanies = [qualityCompaniesPred[k, w] for k in range(numberPlayers)]\n",
-    "    ratingCompanies = [ratingCompaniesPred[k, w] for k in range(numberPlayers)]\n",
-    "    customerScoringQualityRating = np.array([[[(randomCustomerScoringQuality[s][c]*qualityCompanies[k] +\n",
-    "                                                          randomCustomerScoringRating[s][c]*(100-ratingCompanies[k]))for k in range(numberPlayers)] for c in range(numberCustomerEvents)] for s in range(numberSimulations)])\n",
-    "    for a in range(len(prices)):\n",
-    "        AsSim = np.array(As[:,:,w])\n",
-    "        randomCustomerDecision = np.array([[-1 for _ in range(numberCustomerEvents)] for simulationNumber in range(numberSimulations)])\n",
-    "        for s in range(numberSimulations):\n",
-    "            randomCustomerScoringSim = randomCustomerScoring[s]\n",
-    "            pricesCompanies[watchedCompany]=prices[a]\n",
-    "            if numberPlayers>1:\n",
-    "                pricesCompanies[playersWithoutWatched] = [pricesCompaniesPredW[k] for k in playersWithoutWatched]\n",
-    "            numberPriceAdjustmentsBeforeHorizonCurrent = numberPriceAdjustmentsBeforeHorizon[s]\n",
```

Comment:
This large chunk is simply gone as was not moved, right?

Please explain why we can remove it even though another notebook was also removed. Where did all the code go?

Reply:
The code is not simply gone: I replaced the loops with the nested list comprehensions taht were added below. Some chunks were removed because they are not necessary anymore, some calculations were split in the removed code to remove functionality from the innermost loop to maybe speed it up. 

It didn't make it easier to read and understand.

Reply:
By the way: the other notebook was removed because it only created the cond_prob_out.csv which is currently not used in the demand learning notebook. Even if it is needed, we can most likely create it in the monte carlo notebook as well.

Reply:
Ok.

Could you ping me as soon as you have updated the comments + Readme?
The var renaming can be another PR.

Reply:
Ping. 

Changed the readme and also added a good start config(max one minute runtime, not the best results because we now only have 100 simulations but it shows that everything works)+ comment about config.

Reply:
The file that was created in the intensity notebook now gets calculated in the Monte Carlo notebook. Any problems?